### PR TITLE
Bug 1841581: Ensure resolv.conf is in place in prepender before running podman

### PR DESCRIPTION
**Pull request by stbenjam**
Since we now have `set -e`, when podman fails the whole script fails,
and we end up with no resolv.conf at all. Previously, the podman failed
but we fellthrough to where we copy the file to /etc/resolv.conf, and on
the second run it succeeds.

### CODE REVIEW

File: templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml [hunk 1]
```diff
@@ -7,6 +7,10 @@ contents:
     set -eo pipefail
     IFACE=$1
     STATUS=$2
+
+    # Ensure resolv.conf exists before we try to run podman
+    cp /var/run/NetworkManager/resolv.conf /etc/resolv.conf
+
```
Comment (hardys) [inline templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml:13]: It's probably safer to do this in the case I think, so it only copies on a valid $STATUS?

We can then also remove the else "Couldn't find a Virtual IP" part

Comment (stbenjam) [reply templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml]: Ah yea good call

**Comment by hardys** [comment]
lgtm, thanks!

**Pull request merged**